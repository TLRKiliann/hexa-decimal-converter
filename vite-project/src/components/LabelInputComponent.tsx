import React from 'react'

type InputLblProps = {
    value: string | undefined;
    nameValue: string;
    converterTarget: (event: React.ChangeEvent<HTMLInputElement>) => void;
    children: string | JSX.Element | JSX.Element[];
}

export default function LabelInputComponent({value, nameValue, converterTarget, children}: InputLblProps) {
    return (
        <>
            <label>{children} :</label>
            <input
                data-testid="testid-input"
                type="text"
                value={value}
                onChange={(event)=>converterTarget(event)}
                placeholder={nameValue}
            />
        </>
    )
}
