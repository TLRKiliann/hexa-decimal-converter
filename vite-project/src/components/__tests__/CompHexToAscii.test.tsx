import { ChangeEvent } from "react"
import { test, expect, describe } from "vitest"
import { fireEvent, screen, render } from '@testing-library/react'
import { create } from "react-test-renderer"
import "@testing-library/jest-dom"
import CompHexToAscii from '../CompHexToAscii.tsx'
import LabelInputComponent from '../CompHexToAscii.tsx'

test('Snapshat of CompHexToAscii', () => {
    const tree = create(<CompHexToAscii />)
    expect(tree.toJSON()).toMatchSnapshot()
});

describe("captures event from FirstComp", async () => {
  test('testing function converterTarget', () => {
    function converterTarget(evt: ChangeEvent<HTMLInputElement>) {
      expect(evt.target.value).toEqual("changedvalue");
    }
    render(
      <LabelInputComponent>
        <input
          data-testid="testid--input"
          type="text"
          value=""
          onChange={(evt: ChangeEvent<HTMLInputElement>) => converterTarget(evt)}
        />
      </LabelInputComponent>
    )
    const node = screen.getByTestId("testid--input");
    fireEvent.change(node, { target: { value: "changedvalue" }});
    expect(converterTarget).toBeDefined();
  })
});

test('handleClick should update the result', () => {
  const { getByText, getByTestId } = render(<CompHexToAscii />);
  const button = getByText(/Convert/i); 
  fireEvent.click(button);
  const resultElement = getByTestId('span-testid');
  expect(resultElement.textContent).toBe('Result : \u0000');
});