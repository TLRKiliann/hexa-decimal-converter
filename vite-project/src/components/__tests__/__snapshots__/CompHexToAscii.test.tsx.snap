// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshat of CompHexToAscii 1`] = `
[
  <div
    className="labelinput--box"
  >
    <label>
      Hexadecimal
       :
    </label>
    <input
      data-testid="testid--input"
      onChange={[Function]}
      placeholder="Hexadecimal"
      type="text"
      value=""
    />
  </div>,
  <button
    data-testid="testid-btn"
    onClick={[Function]}
    type="button"
  >
    Convert
  </button>,
  <div
    className="box--result"
  >
    <span
      data-testid="span-testid"
    >
      Result : 
      Not binary !!!
    </span>
  </div>,
]
`;
